src/categories/routes.ts

```diff
@@ -0,0 +1,85 @@
+import express from 'express'
+import { asyncWrap, validateRequest, raise404, raise400 } from '../utils'
+import { listCategories, createCategory, getCategory, modifyCategory, deleteCategory } from './service'
+import { categorySchemaValidator, updateCategorySchemaValidator } from './validations'
+import { getCategoryByName } from './models'
+
+const router = express.Router()
+
+router.get('/categories', asyncWrap(async (req, res) => {
+  const categories = await listCategories()
+  res.status(200).json(categories)
+}))
+
+router.get('/categories/:categoryId', asyncWrap(async (req, res) => {
+  const id = parseInt(req.params.categoryId)
+  const category = await getCategory({
+    id: id
+  })
+  if (category === undefined) {
+    res.status(404).json(raise404())
+  } else {
+    res.status(200).json(category)
+  }
+}))
+
+router.post('/categories/byName', asyncWrap(async (req, res) => {
```

Comment:
This should be a GET endpoint and receive a query param indicating the name.

Reply:
Fixed in 5703af4 and fdc5a9b

It is now an endpoint of the form:

**GET** /v1/api/categories/byName?name=<string: lookup name>